server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -343,7 +391,8 @@ && componentTemplatesEquals(this.componentTemplates, other.componentTemplates)
             && Objects.equals(this.metadata, other.metadata)
             && Objects.equals(this.dataStreamTemplate, other.dataStreamTemplate)
             && Objects.equals(this.allowAutoCreate, other.allowAutoCreate)
-            && Objects.equals(this.ignoreMissingComponentTemplates, other.ignoreMissingComponentTemplates);
+            && Objects.equals(this.ignoreMissingComponentTemplates, other.ignoreMissingComponentTemplates)
+            && deprecated() == other.deprecated();
```

Comment:
Same comments here about `hashCode` and `equals` using the variable directly rather than resolving it, it risks thinking a change is a noop when it isn't.